pkg/registration/register/interface.go

```diff
@@ -74,26 +73,14 @@ type RegisterDriver interface {
 	InformerHandler(option any) (cache.SharedIndexInformer, factory.EventFilterFunc)
 }
 
-func IsHubKubeConfigValidFunc(driver RegisterDriver, secretOption SecretOption) wait.ConditionWithContextFunc {
-	return func(ctx context.Context) (bool, error) {
-		logger := klog.FromContext(ctx)
-		if _, err := os.Stat(secretOption.HubKubeconfigFile); os.IsNotExist(err) {
-			logger.V(4).Info("Kubeconfig file not found", "kubeconfigPath", secretOption.HubKubeconfigFile)
-			return false, nil
-		}
-
-		// create a kubeconfig with references to the key/cert files in the same secret
-		hubKubeconfig, err := clientcmd.LoadFromFile(secretOption.HubKubeconfigFile)
-		if err != nil {
-			return false, err
-		}
-
-		if secretOption.BootStrapKubeConfig != nil {
-			if valid, err := IsHubKubeconfigValid(secretOption.BootStrapKubeConfig, hubKubeconfig); !valid || err != nil {
-				return valid, err
-			}
-		}
-
-		return driver.IsHubKubeConfigValid(ctx, secretOption)
-	}
+// Approvers is the inteface that each driver should implement on hub side. The hub controller will use this driver
+// to check the registration request from the agent and cleanup.
+type Approver interface {
+	// Run starts a reconciler on the hub side to monitor the registration request and approve the request
+	// if necessary.
+	Run(ctx context.Context, workers int)
```

Comment:
should we mention that the `Run` method will block and not return immediately?
For me looking at the interface it is easy to think of Run and Cleanup are similar as one-time operations.

Reply:
updated